### PR TITLE
[regression] Added new benchmarks extracted from C++11 book

### PR DESCRIPTION
**Pull request by kunjsong01**
Extracted benchmarks from _C++ How to Program 9th edition_ (C++11). The README file contains test case descriptions and the corresponding language features. 

The new benchmark will be used as part of the TDD process to bring up the clang-cpp-frontend for C++11 verification. 

Done ch1, ch2, ch3 for basic classes and member function calls. 

TODO: 
1. More control statements: ch4, ch5, ~ch6 (ignored, as function overloading and templates not covered in Milestone#2)~  (Done)
2. Pointers and arrays ch7, ch8 (in another PR)

### CODE REVIEW

File: regression/CMakeLists.txt [hunk 1]
```diff
@@ -31,7 +31,7 @@ elseif(WIN32)
     # FUTURE: Add floats-regression esbmc-cpp/cpp
     set(REGRESSIONS esbmc cbmc cstd llvm floats  k-induction )
 else()
-    set(REGRESSIONS esbmc-unix esbmc-unix2 esbmc esbmc-solidity esbmc-old cbmc cstd llvm floats floats-regression k-induction esbmc-cpp/cpp esbmc-cpp/cbmc csmith k-induction-parallel nonz3 bitwuzla incremental-smt)
+    set(REGRESSIONS esbmc-unix esbmc-unix2 esbmc esbmc-solidity esbmc-old cbmc cstd llvm floats floats-regression k-induction esbmc-cpp/cpp esbmc-cpp/cbmc csmith k-induction-parallel nonz3 bitwuzla incremental-smt esbmc-cpp11)
```
Comment (rafaelsamenezes) [inline regression/CMakeLists.txt:34]: You should add this regression for Windows and macOS as well

File: regression/esbmc-cpp11/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Benchmark Description
+
+| Test Case| Main Features Tested |
+| :----:   | :---                 |
```
Comment (rafaelsamenezes) [inline regression/esbmc-cpp11/README.md]: This line seems strange

**Review by rafaelsamenezes — CHANGES_REQUESTED** [review]
Just some nitpicks.

Are you going to add assertions later?

File: regression/esbmc-cpp11/README.md (hunk 2 above)
Comment (kunjsong01) [reply regression/esbmc-cpp11/README.md]: They are for markdown table text alignments.

**Comment by kunjsong01** [comment]
> Are you going to add assertions later?

Yes, I'll add some assertions later to have some "VERIFICATION FAILED" cases. Currently, they only aim to maximise the feature coverage.

File: regression/CMakeLists.txt (hunk 1 above)
Comment (kunjsong01) [reply regression/CMakeLists.txt:34]: Done.

**Review by rafaelsamenezes — APPROVED** [review]

**Review by lucasccordeiro — APPROVED** [review]

**Comment by kunjsong01** [comment]
Done ch1-5. 
Since we've got enough test cases, this PR is ready to merge. I'm going to continue to extend the clang-cpp-frontend to pass these test cases. 

### Notes for C++11:
The test cases in this PR covers two features in C++11:
1. list initialisation 
2. in-class initialisation

**Pull request merged**